### PR TITLE
support http2

### LINKED ISSUE #81: http 2 doesn't get recognized
It falls back to 1.1.
https://github.com/fruux/sabre-http/blob/master/lib/Sapi.php#L134-L136

Happy to provide a pr when you want it to be fixed.

### PR DESCRIPTION
**Pull request by jens1o**
I added the binaries files, that will be installed once you install the dependencies, to the `gitignore` file.

fixes #81

### CODE REVIEW

**Comment by jens1o** [comment]
@staabm Can you help me fix that error, please?

File: tests/HTTP/SapiTest.php [hunk 1]
```diff
@@ -140,6 +140,18 @@ function testSendLimitedByContentLengthString() {
     }
 
     /**
+     * Tests whether http2 is recognized.
+     */
+    function testRecognizeHttp2() {
+        $request = Sapi::createFromServerArray([
+            'SERVER_PROTOCOL' => 'HTTP/2.0',
+            'REQUEST_URI'     => 'bla'
```
Comment (staabm) [inline tests/HTTP/SapiTest.php]: Misses REQUEST_METHOD

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,8 @@ bin/sabre-cs-fixer
 
 # Vim
 .*.swp
+
+# Binaries
+bin/php-cs-fixer*
+bin/phpunit*
+bin/sabre-cs-fixer*
```
Comment (staabm) [inline .gitignore]: Those are already contained, see above

Comment (jens1o) [inline .gitignore]: But the .bat equalivants aren't. I think I'll fix it.

File: tests/HTTP/SapiTest.php (hunk 1 above)
Comment (jens1o) [inline tests/HTTP/SapiTest.php]: fixed.

**Comment by jens1o** [comment]
Why are the tests fill failing?

**Comment by staabm** [comment]
see https://travis-ci.org/fruux/sabre-http/jobs/257649768

seems to be a phpunit issue, unrelated to your PR.

File: .gitignore [hunk 3]
```diff
@@ -6,10 +6,10 @@ composer.lock
 tests/cov/
 
 # Composer binaries
-bin/phpunit
```
Comment (staabm) [inline .gitignore]: please drop all your changes from the gitignore

**Comment by jens1o** [comment]
So you're going to merge it soon?

**Comment by staabm** [comment]
yep

File: .gitignore (hunk 3 above)
Comment (jens1o) [inline .gitignore]: But then I'm going to commit the .bat binaries.

Comment (jens1o) [inline .gitignore]: This prevents that I'm committing these files:
image

Comment (staabm) [inline .gitignore]: ahh I see, you are on windows.

**Comment by staabm** [comment]
thx

**Comment by jens1o** [comment]
When will you publish a new version, I need this feature 😆

**Comment by staabm** [comment]
as I read the trunk changelog, it will be version 5. as this is the only one which will allow BC breaks I need to wait for @evert and what he likes to break in this release.

alternatively we could cherry pick into a 4.x version. lets see what his opinion is.

**Comment by jens1o** [comment]
Okay, thanks.

**Pull request merged**